crates/bevy_ecs/src/intern.rs

```diff
@@ -4,9 +4,15 @@
 //! speed up code by shrinking the stack size of large types,
 //! and make comparisons for any type as fast as integers.
 
+use alloc::{borrow::ToOwned, boxed::Box};
 use core::{fmt::Debug, hash::Hash, ops::Deref};
+
+#[cfg(feature = "std")]
 use std::sync::{OnceLock, PoisonError, RwLock};
 
+#[cfg(not(feature = "std"))]
+use spin::{once::Once as OnceLock, rwlock::RwLock};
```

Comment:
A handful of sync primitives are conditionally replaced with alternatives from `spin`. I would like to revisit our reliance on `spin` at a later date. Spinning is considered a last-resort sync option, and designing it out may improve `std` performance too.